lasagne/tests/layers/test_pool.py

```diff
@@ -116,6 +116,38 @@ def upscale_2d(data, scale_factor):
     return upscaled
 
 
+def spatial_pool(data, pool_dims):
+    input_dim_2 = np.shape(data)[2]
+    input_dim_3 = np.shape(data)[3]
+    pooled_data_list = []
+    num_features = 0
+    for pool_dim in pool_dims:
+        pool_size_2 = input_dim_2 + pool_dim - 1
+        pool_size_2 = pool_size_2 // pool_dim
+
+        pool_size_3 = input_dim_3 + pool_dim - 1
+        pool_size_3 = pool_size_3 // pool_dim
+
+        stride_2 = input_dim_2 // pool_dim
+        stride_3 = input_dim_3 // pool_dim
+
+        pooled_part = max_pool_2d_ignoreborder(
+            data,
+            (pool_size_2, pool_size_3),
+            (stride_2, stride_3),
+            (0, 0)
+        )
```

Comment:
We don't have this indentation style anywhere else. We'd probably do (along with my suggested change above):
```python
pooled_part = max_pool_2d_ignoreborder(
        data, pool_size, stride, (0, 0))
```
I don't have strong feelings about this, though.